File: _tbnf/src/op.py
from __future__ import annotations
from .exceptions import NoBaseName
from .grammar import monot

def basename(t_mut: monot) -> str:
    while True:
        (t,) = (t_mut,)
        if ((t.tag == 3) or (t.tag == 0)) or (t.tag == 4):
            raise NoBaseName(t)
        
        elif t.tag == 2:
            t_mut = t.fields[0]
            continue
        
        else: 
            return t.fields[0]
        
        break

File: tbnf-js/src/src/Op.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.basename = basename;

var _Exceptions = require("./Exceptions.js");

function basename(t_mut) {
  basename: while (true) {
    const t = t_mut;

    switch (t.tag) {
      case 3:
      case 0:
      case 4:
        {
          const exn = new _Exceptions.NoBaseName(t);
          throw exn;
        }

      case 2:
        {
          t_mut = t.fields[0];
          continue basename;
        }

      default:
        {
          return t.fields[0];
        }
    }

    break;
  }
}